google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/grafana/GrafanaCredentialTester.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.testers.grafana;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.tsunami.common.data.NetworkEndpointUtils;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.provider.TestCredential;
+import com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.tester.CredentialTester;
+import com.google.tsunami.proto.NetworkService;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/** Credential tester specifically for grafana. */
+public final class GrafanaCredentialTester extends CredentialTester {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private final HttpClient httpClient;
+
+  private static final String GRAFANA_SERVICE =  "grafana";
+  private static final String GRAFANA_PAGE_TITLE =  "grafana";
+
+  @Inject
+  GrafanaCredentialTester(HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient);
+  }
+
+  @Override
+  public String name() {
+    return "GrafanaCredentialTester";
+  }
+
+  @Override
+  public String description() {
+    return "Grafana credential tester.";
+  }
+
+  /**
+   * Determines if this tester can accept the {@link NetworkService} based on the name of the service or a custom fingerprint.
+   * The fingerprint is necessary since nmap doesn't recognize a grafana instance correctly.
+   * @param networkService the network service passed by tsunami
+   * @return true if a grafana instance is recognized
+   */
+  @Override
+  public boolean canAccept(NetworkService networkService) {
+
+    String uriAuthority = NetworkEndpointUtils.toUriAuthority(networkService.getNetworkEndpoint());
+
+    boolean canAcceptByNmapReport = NetworkServiceUtils.getWebServiceName(networkService).equals(GRAFANA_SERVICE);
+    boolean canAcceptByCustomFingerprint = false;
+
+    var url = String.format("http://%s/", uriAuthority);
```

Comment:
The service could be using HTTPS instead of HTTP so hardcoding 'http' might backfire. Perhaps you could use a  method similar to what I used in my plugin to attempt to determine the correct protocol:

```java
  private static String buildTargetUrl(NetworkService networkService, String path) {
    StringBuilder targetUrlBuilder = new StringBuilder();

    if (NetworkServiceUtils.isWebService(networkService)) {
      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));

    } else {
      // Default to HTTP protocol when the scanner cannot identify the actual service.
      // HTTP is also used in a default Selenium Grid install.
      targetUrlBuilder
          .append("http://")
          .append(toUriAuthority(networkService.getNetworkEndpoint()))
          .append("/");
    }
    targetUrlBuilder.append(path);
    return targetUrlBuilder.toString();
  }
```

Reply:
Fixed in  ffb95c4